libpysal/weights/contiguity.py

```diff
@@ -203,6 +203,13 @@ def from_xarray(cls, da, layer=None, dims={}, sparse=False, **kwargs):
             Default is {} empty dictionary.
         sparse : boolean
             type of weight object. Default is False. For sparse, sparse = True
+        k : int
```

Comment:
I think I'd call this `order` or `orders`. `k` is very aligned with the KNN logic and this is not exactly the same.